### ISSUE TITLE
Ruff does not detect custom OpenCV installation

### ISSUE THREAD

**Issue by Gibtnix** [astral-sh/ruff-vscode#649]
Hello,

First of all, great work, works great and very responsive!

Still, I noticed a minor issue with Python's OpenCV package. On Linux, Ruff does not detect a custom OpenCV build. If I install the Debian package `python3-opencv` (which only installs a `cv2.cpython-XXX-x86_64-linux-gnu.so` file in `/usr/lib/python3/dist-packages/`), Ruff detects and lints OpenCV functions. However, custom compiling OpenCV yields a folder `/usr/local/lib/python3.12/dist-packages/cv2/python-3.12/` with some initialization scripts and a similar `.so` file. If I delete the `__init__.py` file and copy/move only the `.so` to one of the Python module paths, importing and linting with Ruff works. Maybe this can be fixed without moving the installation around? Would be great.

Surprisingly, on Windows linting works even with the `cv2` directory containing the `__init__.py`, i.e. this behavior seems to be platform-dependent.

**Comment by dhruvmanila** [comment]
Can you specify more details about the issue that you're facing? I'm interested in what do you mean by "Ruff detects and lints OpenCV functions". I don't think Ruff has the ability to detect third party packages as it only looks at one file at a time. Could it be that it's a different VS Code extension that's causing the issue?

**Comment by Gibtnix** [comment]
Yes, of course. I mean that linting/autocomplete simply does not work. For example, after writing
```
import numpy as np
np.a...
```
provides autocomplete suggestions as `np.array` etc. as expected. However,
```
import cv2
cv2.im...
```
does not provide autocomplete suggestions as, for example, `cv2.imread`, `cv2.imwrite` etc.
It works, however, if I move the `.so` file and delete the `cv2/__init__.py` script.

**Comment by charliermarsh** [comment]
Ruff doesn't provide autocompletion -- that's provided by a separate language server, like Pylance.

**Comment by Gibtnix** [comment]
Excuse me, thought it would be Ruff that's doing the autocompletion. Sorry for false reporting and thanks for clarifying 👍
